datafusion/physical-expr/src/equivalence/class.rs

```diff
@@ -216,6 +216,17 @@ pub fn const_exprs_contains(
         .any(|const_expr| const_expr.expr.eq(expr))
 }
 
+/// Checks whether `expr` is among in the uniform `const_exprs`.
+pub fn uniform_const_exprs_contains(
+    const_exprs: &[ConstExpr],
+    expr: &Arc<dyn PhysicalExpr>,
+) -> bool {
+    const_exprs.iter().any(|const_expr| {
+        const_expr.expr.eq(expr)
+            && const_expr.across_partitions() != AcrossPartitions::Heterogeneous
```

Comment:
I would find this code more explicit it it checked for `== AcrossPartitions::Uniform` rather than `!= Heterogenious`

Reply:
Something like this (which is how other parts of the code check this):

```diff
index 1e1266b7a..e9e7eb2ca 100644
--- a/datafusion/physical-expr/src/equivalence/class.rs
+++ b/datafusion/physical-expr/src/equivalence/class.rs
@@ -223,7 +223,10 @@ pub fn uniform_const_exprs_contains(
 ) -> bool {
     const_exprs.iter().any(|const_expr| {
         const_expr.expr.eq(expr)
-            && const_expr.across_partitions() != AcrossPartitions::Heterogeneous
+            && matches!(
+                const_expr.across_partitions(),
+                AcrossPartitions::Uniform { .. }
+            )
     })
 }
```